### DISCUSSION TITLE
What is official procedure to update multiple clients after Taskserver issue

### DISCUSSION THREAD

**Discussion by bradyt** [GothenburgBitFactory/taskwarrior#2740] [Q&A]
It is not uncommon for users to go through the following:

1. Setup an account on a Taskserver
2. Configure more than one client towards that account, and sync both of them
3. Have some issue on Taskserver, which either resets tx.data, or user creates new account
4. If new account, reconfigures clients
5. In summary, user now has sync keys in .task/backlog.data that do not exist in tx.data
6. User tries to sync, and gets error, "500 Client sync key not found"
7. User searches internet, either tries `task sync init`, possibly getting "500 Could not find common ancestor for <uuid>", or deletes .task/backlog.data, and reports success

I'm not aware that Taskwarrior team offers official advice on the matter. Is there any procedure here that should be "supported"?

This looks like the closest fit:

- https://github.com/GothenburgBitFactory/taskwarrior/issues/1663

**Comment by djmitche** [comment]
I don't know what the specific issues are here, but in general when things get messed up with taskserver, my approach is to generate a new user (and keys) and sync to that, and leave the old one.

I agree this isn't ideal!

**Reply by bradyt** [reply]
They may have more than one client, and those clients may have distinct unsynced tasks, and non-empty backlog.data files. Does your advice apply in this situation?

**Reply by djmitche** [reply]
No, in my experience that requires manual syncing.

**Reply by djmitche** [reply]
..and to be clear I'm speaking from my own experience -- it's not advice :)

**Reply by bradyt** [reply]
By manual syncing, you mean potentially sorting by modified date, exporting, and reviewing by hand, to find what's missing from the second or nth client?

**Reply by djmitche** [reply]
Yeah, I've also sometimes just looked at backlog.data and re-added those tasks.

**Reply by bradyt** [reply]
>> or user creates new account
>
> when things get messed up with taskserver, my approach is to generate a new user

Hmm. Same thing?

**Reply by djmitche** [reply]
I guess in that case I'd delete backlog.data, since it's relevant to a backlog for a user on the server that's no longer used.

**Reply by bradyt** [reply]
I think I've illustrated scenarios where only applying the backlog.data can result in missing tasks. It's sort of an edge case, but it's still an exception, and number of missing tasks can be arbitrary.

I think you can send tasks to server, they not arrive at second client, have an issue on server, discard the account and create a new one. Those updates are now no longer in the backlog.data of the only client that has that update. So reviewing all tasks by modified date seems safer.

**Reply by djmitche** [reply]
Makes sense to me.. it depends on your tolerance for lost data, I think.

**Reply by bradyt** [reply]
Right, so an "official advice" probably does not recommend to delete backlog.data, and wouldn't suggest running `task sync init` from more than one client (since currently not supported by Taskserver), and currently, may recommend to sort second client by modified, and export and review manually. Assuming that doesn't miss an edge case we haven't thought of yet.
